test/join.jl

```diff
@@ -578,4 +578,23 @@ module TestJoin
         @test eltypes(join(l, r, on=:b, kind=:outer, makeunique=true)) ==
             [Union{Int, Missing}, CS, Union{Int, Missing}]
     end
+
+    @testset "test checks of merge key uniqueness" begin
```

Comment:
Can you also test what happens when the keys contain missing values? What are we supposed to do in that case (and what does Stata do)?

Reply:
test added